suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,235 @@
+        showAlert({
+            title: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.title'),
+            description: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionPrimary',
+            ),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+            secondaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondary',
+            ),
+            onPressSecondaryButton: () => {
+                openLink(
+                    translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondaryUrl'),
+                );
+                hideAlert();
+            },
+        });
+    }, [hideAlert, showAlert, translate, openLink]);
+
+    const addCoinAddress = useCallback(
+        ({ network, type }: { network: Network; type?: AccountType }) => {
+            if (!device?.state) {
+                return false;
+            }
+
+            const accountType = type ?? getDefaultAccountType({ network });
+
+            const currentAccountTypeAccounts = accounts.filter(
+                account => account.symbol === network.symbol && account.accountType === accountType,
+            );
+
+            // Do not allow adding more than 10 accounts of the same type
+            if (currentAccountTypeAccounts.length > LIMIT) {
+                showTooManyAccountsAlert();
+                return false;
+            }
+
+            // Do not allow adding another empty account if there is already one
+            const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
+
+            if (emptyAccounts.length > 0) {
+                showAnotherEmptyAccountAlert();
+                return false;
+            }
+
+            const addNetworkAccount = async (deviceState: string) => {
+                setIsAddingCoinAccount(true);
+                const result = await dispatch(
+                    addNetworkAccountThunk({
+                        network,
+                        accountType,
+                        deviceState,
+                    }),
+                ).unwrap();
+
+                setIsAddingCoinAccount(false);
+
+                if (result.account) {
+                    navigation.navigate(RootStackRoutes.AccountDetail, {
+                        accountKey: result.account.key,
+                        tokenContract: undefined,
+                    });
+                }
+            };
+
+            addNetworkAccount(device.state);
+        },
+        [
+            device?.state,
+            getDefaultAccountType,
+            accounts,
+            showTooManyAccountsAlert,
+            showAnotherEmptyAccountAlert,
+            dispatch,
+            navigation,
+        ],
```

Comment:
Nit: As discussed, no functions here need to be `useCallback` and the deps then can be removed as well. Up to you if you want to change it or not. Feel free to resolve

Reply:
updated in 5106077413

Reply:
IMO any function that is returned from a hook should be wrapped in the `useCallback`. Like that you can prevent unnecessary rerenders while accessing the function in a component that uses the hook.

if you don't wrap in in a `useCallback` the return value of the hook is changed on every execution of the hook function and the component is rerendered.

Reply:
IMO unnecessary. This hook is used only once and if it rerenders, the whole state gets reset as well, which will cause this hook to rerender as well. Without usage of memoize, this useCallback is not bringing any performance improvements and is redundant. Same as with selectors, optimize when necessary I guess. Not everywhere by default.